src/main/java/com/google/devtools/build/lib/actions/FileArtifactValue.java

```diff
@@ -123,6 +123,14 @@ public boolean isRemote() {
     return false;
   }
 
+  /**
+   * Returns {@code true} if the file is remote and is available remotely at time {@code epochMilli}
+   * which is the milliseconds since epoch.
+   */
+  public boolean isRemotelyAvailable(long epochMilli) {
```

Comment:
Do we ever call that on a FileArtifactValue for which we don't know it is a RemoteFileArtifactValue?

Reply:
No. I can move this into `RemoteFileArtifactValue`.